src/plugins/bfetch/server/plugin.ts

```diff
@@ -129,10 +142,23 @@ export class BfetchServerPlugin
 
   public stop() {}
 
+  private async getCompressionDisabled(
+    request: KibanaRequest,
+    getStartServices: StartServicesAccessor
+  ) {
+    const [core] = await getStartServices();
+    const uiSettingsClient = core.uiSettings.asScopedToClient(
+      core.savedObjects.getScopedClient(request)
+    );
+    return uiSettingsClient.get(DISABLE_SEARCH_COMPRESSION);
```

Comment:
I understand this is needed because pr doesn't follow https://github.com/elastic/kibana/pull/98658 . 

If we'd go with it, then I think, it would be better if client dictates if it wants encoded response. We could probably use native 'Accept` header for this, similar to how native encoding works. 

We also would save a potential trip for a uiSetting on a server. 

I think there is also a possible bug now, that client may have one setting cached, but the setting may has been already been changed on that server. And in that case client will break until the page is reloaded.

Reply:
I tried using `Accept-Encoding` and `Content-Encoding` headers.
The browser [doesn't allow](https://stackoverflow.com/questions/32463340/angular-gzip-json-file-automatically-refused-to-set-unsafe-header-accept-enco) you to set `Accept-Encoding` at all.
And if you set `Content-Encoding` to `deflate`, the browser itself tries to inflate the response itself, and obviously fails.

Reply:
Agreed about the bug.

Reply:
Fixed